manifests/ssltools/nssdb.pp

```diff
@@ -1,50 +1,27 @@
 # Sets up nssdb
 class certs::ssltools::nssdb (
   $nss_db_dir = $certs::nss_db_dir,
+  $nss_db_password_path = $certs::nss_db_password_path,
+  $nss_db_password = $certs::nss_db_password,
   $group = 'qpidd',
 )  {
   Exec { logoutput => 'on_failure' }
 
-  $nss_db_password_file   = "${nss_db_dir}/nss_db_password-file"
```

Comment:
Unfortunately ,changing this to be the file itself and not the path will break things:

https://github.com/theforeman/puppet-katello/blob/master/manifests/qpid.pp#L23-L25

Reply:
As we undertake this work of building better tools to manage certificates for a more flexible, modular, and maintainable design, I'd like to standardize as much as possible the naming conventions for variables between the various providers and manifests here.

So with `keystore_certificate` we have an established convention that `name` refers to the file basename and `path` refers to fully qualified path.

I've opened https://github.com/theforeman/puppet-katello/pull/419 to explore what it looks like to go forward with the change.

Reply:
As an aside, should the change be considered "breaking" if it only renames a variable that is not in the declared interface for the class? Is this a case where we should perhaps move that variable into the interface, or is this actually pretty standard practice whenever the class isn't explicitly declared "private" ?

Reply:
I'm glad you asked, cause I also dont know that I am clear on the answer between variables declared as parameters vs. variables found in the class itself. Help us @ekohl you're our only hope

Reply:
> As we undertake this work of building better tools to manage certificates for a more flexible, modular, and maintainable design, I'd like to standardize as much as possible the naming conventions for variables between the various providers and manifests here.
> 
> So with `keystore_certificate` we have an established convention that `name` refers to the file basename and `path` refers to fully qualified path.

I think the care that should be taken is not mixing the two concerns: creating a new provider and renaming variables. Both are valid, and can be done independently of one another. So if you think we should rename, lets do that as its own PR and change and then build on that. Or tackle it as a follow up. This makes the changes easier to understand, review and also coordinate the fixes of across modules.

If we aim to push the management of the keystore, truststore and nssdb into the modules that use them (e.g. puppet-candlepin, puppet-qpid) I am not too worried about naming conventions in this module as we will be doing a larger change and can ensure we follow convention there.

Reply:
> I'm glad you asked, cause I also dont know that I am clear on the answer between variables declared as parameters vs. variables found in the class itself. Help us @ekohl you're our only hope

Thinking about it a bit more, I guess the answer is obvious: declared parameters are inputs to the class which can be (or are required to be) assigned a value. Variables which are not declared as parameters could still be used elsewhere, but you don't have the ability to directly assign different values to them.

Reply:
I think variables declared inside a class is a bit of an odd thing in versioning. AFAIK it's not really described in documentation and as such not really an API (as opposed to class parameters). However, Puppet doesn't have a concept of private variables so other classes can use them (and we do - it can be very useful).

Maybe one solution is to always consider them private unless otherwise declared. That declaration could be done in puppet-strings using `@example` blocks. Really doing it well would probably mean introducing explicit `@variable` tags in puppet-strings.

That said, we don't make a statement about the API of this class. Perhaps this whole file should be considered private and add `@api private`.